bindings/java/src/main/java/org/openucx/jucx/ucp/UcpWorker.java

```diff
@@ -83,13 +83,23 @@ public long getMaxAmHeaderSize() {
      *                   this value should be set to null.
      */
     public void setAmRecvHandler(int amId, UcpAmRecvCallback callback) {
-        if (callback != null) {
-            amRecvHandlers.put(amId, callback);
-            callback.setWorker(this);
-        } else {
-            amRecvHandlers.remove(amId);
+        if (callback == null) {
+            removeAmRecvHandler(amId);
+            return;
         }
-        setAmRecvHandlerNative(getNativeId(), amId, callback);
+        Object[] callbackAndWorker = new Object[2];
```

Comment:
maybe use Pair?

Reply:
There's no such class in java. To not create one just for this case, decided just to use an array of 2 elements